DuckDuckGo/MaliciousSiteProtection/Services/MaliciousSiteProtectionDatasetsFetcher.swift

```diff
@@ -0,0 +1,303 @@
+//
+//  MaliciousSiteProtectionDatasetsFetcher.swift
+//  DuckDuckGo
+//
+//  Copyright © 2025 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import MaliciousSiteProtection
+import BackgroundTasks
+import Core
+import Combine
+
+protocol MaliciousSiteProtectionDatasetsFetching {
+    func startFetching()
+    func registerBackgroundRefreshTaskHandler()
+}
+
+final class MaliciousSiteProtectionDatasetsFetcher: MaliciousSiteProtectionDatasetsFetching {
+    private let featureFlagger: MaliciousSiteProtectionFeatureFlagger & MaliciousSiteProtectionFeatureFlagsSettingsProvider
+    private let userPreferencesManager: MaliciousSiteProtectionPreferencesProvider
+    private let dateProvider: () -> Date
+    private let updateManager: MaliciousSiteUpdateManaging
+    private let backgroundTaskScheduler: BGTaskScheduling
+    private let application: BackgroundRefreshCapable
+
+    private var preferencesManagerCancellable: AnyCancellable?
+
+    private var shouldUpdateHashPrefixSets: Bool {
+        dateProvider().timeIntervalSince(updateManager.lastHashPrefixSetUpdateDate) > .minutes(featureFlagger.hashPrefixUpdateFrequency)
```

Comment:
not: should we instead check the file modification date on the FS. I‘ve noticed it won‘t update the files after I remove them manually because these values are show it‘s up-to-date 🤔